src/main/java/teammates/common/util/StringHelper.java

```diff
@@ -359,6 +359,104 @@ public static String csvToHtmlTable(String str) {
                              result.toString());
     }
 
+    private static List<String> combineRowDataEntries(List<String> rowData,
+            String delimiter) {
+        String combinedEntries = rowData.get(0) + delimiter + rowData.get(1);
+        rowData = new ArrayList<String>();
+        rowData.add(combinedEntries);
+        return rowData;
+    }
+
+    /**
+     * Convert a csv string to a beautified html table string for displaying
+     * 
+     * @param str
+     * @return beautified html table string
+     */
+    public static String csvToBeautifiedHtmlTable(String str) {
+
+        str = handleNewLine(str);
+        String[] lines = str.split(Const.EOL);
+
+        StringBuilder result = new StringBuilder();
+        int globalMaxLength = 0; // global max. no. of columns
+        int qnMaxLength = 0; // max. no. of columns in a question
+
+        List<ArrayList<String>> rowDataArray = new ArrayList<ArrayList<String>>();
+        List<Integer> qnMaxLengthArray = new ArrayList<Integer>();
+
+        for (int i = 0; i < lines.length; i++) {
+
+            List<String> rowData = getTableData(lines[i]);
+
+            if (checkIfEmptyRow(rowData)) {
+                continue;
+            }
+            if (rowData.get(0).equals("Course") ||
+                    rowData.get(0).equals("Session Name")) {
```

Comment:
Magic string. This can break if the heading was changed later. May be unify using constants?